src/main/java/duke/tasks/TaskList.java

```diff
@@ -0,0 +1,306 @@
+        EVENT
+    }
+
+    // Storage to save and load tasks from disk.
+    private final Storage taskStorage;
+
+    /**
+     * Constructor for a TaskList.
+     *
+     * @param saveFileName The name of the file to save the data to. Eg: tasks.txt
+     */
+    public TaskList(String saveFileName) {
+        this.taskStorage = new Storage(saveFileName);
+        this.list = new ArrayList<>();
+        this.importData();
+    }
+
+    // Returns "task" if there is only 1 task in the list, else returns "tasks".
+    private String taskOrTasks() {
+        return this.list.size() == 1 ? "task" : "tasks";
+    }
+
+    /**
+     * Adds a task to the TaskList. This method is private.
+     *
+     * @param taskType The type of task to add.
+     * @param description The description of the task to add.
+     * @param by The deadline of the task to add.
+     * @param start The start date/time of the task to add.
+     * @param end The end date/time of the task to add.
+     * @return String message
+     */
+    private String add(TaskType taskType, String description, String by, String start, String end)
+            throws DukeIllegalArgumentException {
+
+        // Avoids early return statements in switch statement.
+        String output = "";
+
+        switch (taskType) {
+        case TODO:
+            Task toDoTask = new ToDo(description);
+            this.list.add(toDoTask);
+            this.exportData();
+            output = String.format(ADD_TASK_TEMPLATE,
+                    toDoTask, this.list.size(), taskOrTasks());
+            break;
+        case DEADLINE:
+            Task deadlineTask = new Deadline(description, by);
+            this.list.add(deadlineTask);
+            this.exportData();
+            output = String.format(ADD_TASK_TEMPLATE,
+                    deadlineTask, this.list.size(), taskOrTasks());
+            break;
+        case EVENT:
+            Task eventTask = new Event(description, start, end);
+            this.list.add(eventTask);
+            this.exportData();
+            output = String.format(ADD_TASK_TEMPLATE,
+                    eventTask, this.list.size(), taskOrTasks());
+            break;
+        default:
+            break;
+        }
+        return output;
+    }
+
+    /**
+     * Adds a task with description to the TaskList. Used for ToDo tasks.
+     *
+     * @param taskType The type of task to add.
+     * @param description The description of the task to add.
+     * @return String message
+     */
+    public String add(TaskType taskType, String description) throws DukeIllegalArgumentException {
+        if (description.isBlank()) {
+            throw new DukeIllegalArgumentException("The description of a ToDo task cannot be blank.");
+        }
+        if (taskType != TaskType.TODO) {
+            throw new DukeIllegalArgumentException("Only ToDo tasks can be added with just a description.");
+        }
+        return this.add(taskType, description, "", "", "");
+    }
```

Comment:
Maybe the methods might be clearer if it was specified which type of Task it accepted in the name?
I understand your approach though, so it's your call on whether this is something that needs to be changed.